src/test/java/com/teragrep/rlp_03/client/ClientTest.java

```diff
@@ -0,0 +1,198 @@
+        RunnableEventLoop runnableEventLoop = new RunnableEventLoop(new EventLoopFactory().create());
+        Thread eventLoopThread = new Thread(runnableEventLoop);
+        eventLoopThread.start();
+
+        // client takes resources from this pool
+        ExecutorService executorService = Executors.newCachedThreadPool();
+
+        // client connection type
+        SocketFactory socketFactory = new PlainFactory();
+
+        ConnectContextFactory connectContextFactory = new ConnectContextFactory(executorService, socketFactory);
+        ClientFactory clientFactory = new ClientFactory(connectContextFactory, runnableEventLoop.eventLoop());
+
+        try (Client client = clientFactory.open(new InetSocketAddress("localhost", port))) {
+
+            // send open
+            CompletableFuture<AbstractMap.SimpleEntry<String, byte[]>> open = client
+                    .transmit("open", "a hallo yo client".getBytes(StandardCharsets.UTF_8));
+
+            // send syslog
+            CompletableFuture<AbstractMap.SimpleEntry<String, byte[]>> syslog = client
+                    .transmit("syslog", "yonnes payload".getBytes(StandardCharsets.UTF_8));
+
+            // send close
+            CompletableFuture<AbstractMap.SimpleEntry<String, byte[]>> close = client
+                    .transmit("close", "".getBytes(StandardCharsets.UTF_8));
+
+            // test open response
+            AbstractMap.SimpleEntry<String, byte[]> openResponse = open.get();
+            LOGGER.debug("openResponse <[{}]>", openResponse);
+            Assertions.assertEquals("rsp", openResponse.getKey());
+            Assertions
+                    .assertEquals(
+                            "200 OK\nrelp_version=0\nrelp_software=RLP-01,1.0.1,https://teragrep.com\ncommands=syslog\n",
+                            new String(openResponse.getValue(), StandardCharsets.UTF_8)
+                    );
+
+            // test syslog response
+            AbstractMap.SimpleEntry<String, byte[]> syslogResponse = syslog.get();
+            Assertions.assertEquals("rsp", syslogResponse.getKey());
+            LOGGER.debug("syslogResponse <[{}]>", syslogResponse);
+            Assertions.assertEquals("200 OK", new String(syslogResponse.getValue(), StandardCharsets.UTF_8));
+
+            // test close response
+            AbstractMap.SimpleEntry<String, byte[]> closeResponse = close.get();
+            Assertions.assertEquals("rsp", closeResponse.getKey());
+            LOGGER.debug("closeResponse <[{}]>", closeResponse);
+            Assertions.assertEquals("", new String(closeResponse.getValue(), StandardCharsets.UTF_8));
+
+            // close the client eventLoop
+            runnableEventLoop.close();
+            eventLoopThread.join();
+        }
+        catch (InterruptedException | ExecutionException | IOException exception) {
+            throw new RuntimeException(exception);
+        }
```

Comment:
I'd add Assertions.fail("IOException happened")

Reply:
will be fixed later #120